### PR TITLE
Live Project Counters (updated)

### PR DESCRIPTION
**Pull request by zwolf**
Live project classification counters update via pusher. A little cleanup required that didn't get done by the end of last ZTM hackday. Using new git functionality, should be possible to commit to my fork!

Branch name is misleading.

Describe your changes.

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch?

## Optional

- [x] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?

### CODE REVIEW

**Comment by srallen** [comment]
Deployed to: https://hackday-chat.pfe-preview.zooniverse.org/projects?env=production

**Comment by marten** [comment]
@rogerhutchings could you lend this your watchful eye?

**Comment by rogerhutchings** [comment]
will do

**Comment by rogerhutchings** [comment]
`app/pages/project/metadata.jsx` should probably try and unsubscribe to the pusher stream when the component is unmounted, I'm getting the following warning (just a warning, though)

```
Pusher : Error : {"type":"WebSocketError","error":{"type":"PusherError","data":{"code":null,"message":"Existing subscription to channel panoptes"}}}
```

File: app/pages/project/metadata.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    return (
+      <div className="project-metadata content-container">
+        <div className="project-metadata-header">
+          <span>{project.display_name}</span>{' '}
+          <Link to={statsLink}>
+            <span>Statistics</span>
+          </Link>
+        </div>
+
+        <div className="project-metadata-stats">
```
Comment (rogerhutchings) [inline app/pages/project/metadata.jsx]: I'd probably abstract these out to a class method at least, if not a component

File: app/pages/project/metadata.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            <div>Retired Subjects</div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+ProjectMetadata.contextTypes = {
+  pusher: React.PropTypes.object,
```
Comment (rogerhutchings) [inline app/pages/project/metadata.jsx]: This should have `.isRequired`

File: app/pages/project/metadata.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+ProjectMetadata.propTypes = {
+  project: React.PropTypes.object,
```
Comment (rogerhutchings) [inline app/pages/project/metadata.jsx]: This should have `.isRequired`

File: app/pages/project/metadata.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+ProjectMetadata.defaultProps = {
+  project: null,
```
Comment (rogerhutchings) [inline app/pages/project/metadata.jsx]: ...but if you do, this would have to be `{}`

File: app/pages/project/metadata.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { Link } from 'react-router';
+
+class ProjectMetadata extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      classificationsCount: this.props.project.classifications_count,
```
Comment (rogerhutchings) [inline app/pages/project/metadata.jsx]: I don't think you need `this` here as props is passed in already

File: app/main.cjsx [hunk 6]
```diff
@@ -30,3 +30,6 @@ require('./lib/log-deployed-commit')()
 
 # Just for console access:
 window?.zooAPI = require 'panoptes-client/lib/api-client'
+
+
+
```
Comment (rogerhutchings) [inline app/main.cjsx]: This feels so much more roomy 😄

File: app/pages/project/metadata.jsx (hunk 4 above)
Comment (rogerhutchings) [inline app/pages/project/metadata.jsx]: I'm actually surprised it doesn't complain already

**Comment by srallen** [comment]
Good catches. Missed some stuff in my haste. I just merged the testing branch, so definitely could get some in for this, though I may not be able to get back to this until later in the week.

**Comment by marten** [comment]
I'm happy to continue the work on this branch too, if you're short on time.

**Comment by srallen** [comment]
@marten thanks. I don't think I'll get to it this week as my Thursday and Friday will be taken mostly up with meetings.

**Comment by marten** [comment]
That's okay, I'll pick it up from here.

**Comment by marten** [comment]
@rogerhutchings Comments addressed and tests added.

**Pull request merged**